File: java-playground/src/test/java/org/gonza/javaplayground/lotto/domain/LottoNumberTest.java
package org.gonza.javaplayground.lotto.domain;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

class LottoNumberTest {

	@Test
	void LottoNumber를_생성할_수_있다() {
		LottoNumber number = LottoNumber.of(1);

		assertEquals(1, number.getValue());
	}

	@Test
	void LottoNumber는_1_미만일_수_없다() {
		assertThrows(IllegalArgumentException.class, () -> LottoNumber.of(0));
	}

	@Test
	void LottoNumber는_45를_초과할_수_없다() {
		assertThrows(IllegalArgumentException.class, () -> LottoNumber.of(46));
	}

	@Test
	void LottoNumber의_value가_동일하다면_같다() {
		LottoNumber number1 = LottoNumber.of(1);
		LottoNumber number2 = LottoNumber.of(1);

		assertEquals(number1, number2);
	}

	@Test
	void LottoNumber의_value가_동일하지않다면_다르다() {
		LottoNumber number1 = LottoNumber.of(1);
		LottoNumber number2 = LottoNumber.of(2);

		assertNotEquals(number1, number2);
	}
}

File: kotlin-playground/src/test/kotlin/org/gonza/kotlinplayground/domain/lotto/LottoNumberTest.kt
package org.gonza.kotlinplayground.domain.lotto

import org.gonza.kotlinplayground.domain.lotto.exception.InvalidLottoNumberException
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class LottoNumberTest {
    @Test
    fun `번호는 음수일 수 없다`() {
        assertThrows(InvalidLottoNumberException::class.java) {
            LottoNumber(-1)
        }
    }

    @Test
    fun `번호는 45를 초과할 수 없다`() {
        assertThrows(InvalidLottoNumberException::class.java) {
            LottoNumber(46)
        }
    }

    @Test
    fun `번호는 값이 같다면 같은 번호이다`() {
        val number = LottoNumber(1)
        val other = LottoNumber(1)

        assertEquals(number, other)
    }
}
